### DISCUSSION TITLE
Undocumented features

### DISCUSSION THREAD

**Discussion by Isty001** [clibs/clib#233] [General]
I want to document the available commands, because some of the are not well known. So currently we have these:

```
    init                 Start a new project
    i, install [name...] Install one or more packages
    up, update [name...] Update one or more packages
    upgrade [version]    Upgrade clib to a specified or latest version
    configure [name...]  Configure one or more packages
    build [name...]      Build one or more packages
    search [query]       Search for packages
    help <cmd>           Display help for cmd
```

However I am not sure how some of them should work. Could you give me some info about what is the intended function of the `update`, `configure` and `build` commands?

**Comment by jwerle** [comment]
IIRC, `clib update` is more or less a modified `clib install` that ensures the cache is skipped, installation is "forced", and global installs are disabled

`clib configure` will execute the `configure` script in a `clib.json` as seen here: https://github.com/jwerle/libmerkle/blob/master/clib.json#L7)

`clib build` will execute `make` for a package with `-C` set to the `Makefile` specified in a `clib.json` as seen here:(https://github.com/jwerle/libmerkle/blob/master/clib.json#L8

**Reply by Isty001** [reply]
Thank you!

**Reply by Isty001** [reply]
Let me know if I forgot something https://github.com/clibs/clib/wiki/Command-Line-Interface
